ddprof-profiles/src/lib.rs

```diff
@@ -763,4 +828,124 @@ mod api_test {
+            },
+        ];
+
+        let mut profile: Profile = Profile::builder().sample_types(sample_types).build();
+
+        let id_label = api::Label {
+            key: "local root span id",
+            str: Some("10"),
+            num: 0,
+            num_unit: None,
+        };
+
+        let id2_label = api::Label {
+            key: "local root span id",
+            str: Some("11"),
+            num: 0,
+            num_unit: None,
+        };
+
+        let other_label = api::Label {
+            key: "other",
+            str: Some("test"),
+            num: 0,
+            num_unit: None,
+        };
+
+        let sample1 = api::Sample {
+            locations: vec![],
+            values: vec![1, 10000],
+            labels: vec![id_label, other_label],
+        };
+
+        let sample2 = api::Sample {
+            locations: vec![],
+            values: vec![1, 10000],
+            labels: vec![id2_label, other_label],
+        };
+
+        profile.add(sample1).expect("add to success");
+
+        profile.add(sample2).expect("add to success");
+
+        profile.add_endpoint(Cow::from("10"), Cow::from("my endpoint"));
+
+        let serialized_profile: pprof::Profile = (&profile).into();
+
+        assert_eq!(serialized_profile.sample.len(), 2);
+
+        let s1 = serialized_profile.sample.get(0).expect("sample");
+
+        // The trace endpoint label should be added to the first sample
+        assert_eq!(s1.label.len(), 3);
+
+        let l1 = s1.label.get(0).expect("label");
+
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l1.key as usize)
+                .unwrap(),
+            "local root span id"
+        );
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l1.str as usize)
+                .unwrap(),
+            "10"
+        );
+
+        let l2 = s1.label.get(1).expect("label");
+
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l2.key as usize)
+                .unwrap(),
+            "other"
+        );
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l2.str as usize)
+                .unwrap(),
+            "test"
+        );
+
+        let l3 = s1.label.get(2).expect("label");
+
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l3.key as usize)
+                .unwrap(),
+            "trace endpoint"
+        );
+        assert_eq!(
+            serialized_profile
+                .string_table
+                .get(l3.str as usize)
+                .unwrap(),
+            "my endpoint"
+        );
+
+        let s2 = serialized_profile.sample.get(1).expect("sample");
+
+        // The trace endpoint label shouldn't be added to second sample because the span id doesn't match
+        assert_eq!(s2.label.len(), 2);
```

Comment:
so when `local root span id` does not match a known endpoint, we remove the label from the serialized data
(just checking if I understand this well)

Reply:
We don't touch `local root span id`, but we only add the endpoint label if `local root span id` matches a known endpoint

Reply:
This is a good point though -- I guess since we `intern(...)` the endpoint, we still ship it, even if there's no matching `local root span id` that needs it.

Which I don't think is particularly problematic (but perhaps documenting with a comment on the API?).

Reply:
Comment added in https://github.com/DataDog/libdatadog/pull/33/commits/f8fd2f958c99d9a75bf0afe2bb492c6d2546ea50